scheduler/replica_scheduler.go

```diff
@@ -52,6 +52,12 @@ func (rcs *ReplicaScheduler) ScheduleReplica(replica *longhorn.Replica, replicas
 		return nil, nil, fmt.Errorf("BUG: Replica %v has been scheduled to node %v", replica.Name, replica.Spec.NodeID)
 	}
 
+	// not to schedule a replica failed and unused before.
```

Comment:
Then will this kind of replica block the replica replenishment? or can these replicas be reused later?

Reply:
No, it will not block the replica replenishment.

Reply:
Can you explain more about it? I don't remember to the details...

Reply:
`replenishReplicas` will get Replenish Replicas Count [here](https://github.com/longhorn/longhorn-manager/blob/master/controller/volume_controller.go#L2112-L2114) (mainly according to `r.Spec.FailedAt == ""`)
and then use '[replica-scheduler.CheckAndReuseFailedReplica](https://github.com/longhorn/longhorn-manager/blob/master/scheduler/replica_scheduler.go#L434)' to get reusable replicas (it should be empty in this case.), finally check if it requires to create new replica.
I think replica replenishment would not get into the codes I add and that's what I saw when I tested the codes.

Reply:
Got it. Since it's not a new volume, the replenishment count should be 1. And the potentially reusable replica is the only replica that contains data. After the salvage or reuse, the volume controller will ignore those failed & unscheduled replicas then directly replenish new replicas, right?

Reply:
Yes, that's right, only replenishment count will be `v.Spec.NumberOfReplicas - usableCount`.